app/src/main/java/org/mozilla/fenix/browser/BaseBrowserFragment.kt

```diff
@@ -306,8 +307,9 @@ abstract class BaseBrowserFragment : Fragment(), UserInteractionHandler, Session
                     sessionManager = sessionManager,
                     sessionId = customTabSessionId,
                     fragmentManager = parentFragmentManager,
-                    launchInApp = { context.settings().preferences.getBoolean(
-                        context.getPreferenceKey(R.string.pref_key_open_links_in_external_app), false)
+                    launchInApp = {
+                        PreferenceManager.getDefaultSharedPreferences(context).getBoolean(
+                            context.getPreferenceKey(R.string.pref_key_open_links_in_external_app), false)
```

Comment:
Could a new `Settings` variable be added for this?

Reply:
Agreed. Thanks for catching this.

Reply:
update, service will still have get the preference through `PreferenceManager` since it have a different context.